platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/daemon/client/DaemonClientInputForwarder.java

```diff
@@ -15,118 +15,124 @@
  */
 package org.gradle.launcher.daemon.client;
 
-import org.gradle.api.logging.Logger;
-import org.gradle.api.logging.Logging;
-import org.gradle.internal.concurrent.ExecutorFactory;
+import org.gradle.internal.UncheckedException;
 import org.gradle.internal.concurrent.Stoppable;
 import org.gradle.internal.dispatch.Dispatch;
-import org.gradle.internal.io.TextStream;
 import org.gradle.internal.logging.console.GlobalUserInputReceiver;
 import org.gradle.internal.logging.console.UserInputReceiver;
 import org.gradle.launcher.daemon.protocol.CloseInput;
 import org.gradle.launcher.daemon.protocol.ForwardInput;
 import org.gradle.launcher.daemon.protocol.InputMessage;
 import org.gradle.launcher.daemon.protocol.UserResponse;
 
-import javax.annotation.Nullable;
+import java.io.BufferedReader;
+import java.io.IOException;
 import java.io.InputStream;
-import java.util.concurrent.atomic.AtomicReference;
+import java.io.InputStreamReader;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
-/**
- * Eagerly consumes from an input stream, sending each line as a commands over the connection and finishing with a {@link CloseInput} command.
- * Each line is forwarded as either a {@link ForwardInput}, for non-interactive text, or a {@link UserResponse}, when expecting some interactive text.
- */
 public class DaemonClientInputForwarder implements Stoppable {
```

Comment:
This is the main client piece. It now reads from stdin only when requested to do so by the daemon.